### ISSUE TITLE
Wrong SQL generated because of positions of parameters within LINQ

### ISSUE THREAD

**Issue by StefanGeorgievski** [JasperFx/marten#871]
Hello everyone. While working with LINQ I noticed something weird. I am using simple example to report the bug that is happening when trying to compare dates in Where statement.
```C#
var limit = DateTime.Now;
            try
            {
                var session = Store.OpenSession();
                var users = await session.Query<User>()
                    .Where(x => limit <= x.Birthday)
                    .ToListAsync();

                var sql = session.Query<User>()
                    .Where(x => limit <= x.Birthday)
                    .ToCommand(FetchType.FetchMany);
                Console.WriteLine(sql.CommandText);

                var sql2 = session.Query<User>()
                    .Where(x => x.Birthday >= limit)
                    .ToCommand(FetchType.FetchMany);
                Console.WriteLine(sql2.CommandText);

                session.Dispose();
            }
            catch
            {
                throw new Exception("Error while working with the db");
            }
```

The SQL generated from the LINQ is different in the two cases even though it should be the same. 
My comparison is basically A >= B which is the same as B <= A.

The first SQL which is wrong is: select d.data, d.id, d.mt_version from public.mt_doc_user as d where public.mt_immutable_timestamp(d.data ->> 'Birthday') <= :arg0
The second SQL which is correct is: select d.data, d.id, d.mt_version from public.mt_doc_user as d where public.mt_immutable_timestamp(d.data ->> 'Birthday') >= :arg0

![image](https://user-images.githubusercontent.com/26904243/30326561-31ed75e4-97c9-11e7-9fb1-6a2e598c39eb.png)

**Comment by jimgolfgti** [comment]
@StefanGeorgievski Could you confirm this is still an issue with the latest master as I think this was addressed in #869 

If so could you close, thx

**Comment by StefanGeorgievski** [comment]
@jimgolfgti I had older version of Marten and updated it to Marten 2.2.0 but the update did not solve the problem. The queries were still wrong.

**Comment by jimgolfgti** [comment]
PR was merged 12 days ago, 2.2 shipped about 22 days ago.

Fix will be out in 2.3

**Comment by StefanGeorgievski** [comment]
Okay than, I am closing this one.

Thanks.
